zebra-network/src/peer_set/initialize.rs

```diff
@@ -285,20 +287,23 @@ where
                 connection_tracker,
             };
 
-            let outbound_connector = outbound_connector.clone();
-            async move {
-                // Rate-limit the connection, sleeping for an interval according
-                // to its index in the list.
+            // Construct a connector future but do not drive it yet ...
+            let outbound_connector_future = outbound_connector
+                .clone()
+                .oneshot(req)
+                .map_err(move |e| (addr, e));
+
+            // ... instead, spawn a new task to handle this connector
+            tokio::spawn(async move {
+                let task = outbound_connector_future.await;
+                // Only spawn one outbound connector per `MIN_PEER_CONNECTION_INTERVAL`.
```

Comment:
Nit: let's explain the multiplication, using a similar comment to the previous code

```suggestion
                // Only spawn one outbound connector per `MIN_PEER_CONNECTION_INTERVAL`,
                // sleeping for an interval according to its index in the list.
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/3189/commits/ec0099fe1a406713b63fd2c18c417cdacf55f737